### ISSUE TITLE
Questions on softmax layer

### ISSUE THREAD

**Issue by ShangxuanWu** [TheLegendAli/DeepLab-Context#12]
Hi!

I am reproducing the results of DeepLab, and I met a problem.

I used the method mentioned [here](https://github.com/TheLegendAli/DeepLab-Context/issues/2), and when I run run.py, it told me that there are unexpected labels in softmax layer (such as 55, 150 and so on). I have to force NUM_LABELS=255 to make it run.

How could I fix this problem? Thanks very much!

Shangxuan

**Comment by TheLegendAli** [comment]
My understanding is that the segment pixels have wrong labeling. Not sure what you did wrong.

easier solution use:

https://www.dropbox.com/s/oeu149j8qtbs1x0/SegmentationClassAug.zip?dl=0

**Comment by SJTUsuperxu** [comment]
hi @TheLegendAli , I use the C++ version of deeplab and this error also troubles me. I want to finetune the deeplab model for object segmentation thus the number of classes should be 2, but there are always unexpected labels in softmax layer. I have checked the segment labels and it indeed has 2 classes. Could you give me some instructions?  Thank you~~

**Comment by SJTUsuperxu** [comment]
hi @ShangxuanWu 
Have you solved the problem??? 
Thank you~~
